File: psets/cash/cash.c
#include<stdio.h>
#include<cs50.h>
#include<math.h>

float get_unsigned_float(string str);
int minimize(int n);
int number_of_coins(int cash, int coin);

int main(void)
{
    float dollers;
    int cent;
    dollers = get_unsigned_float("Change: ");
    cent = round(dollers * 100);
    printf("%i\n", minimize(cent));
}

float get_unsigned_float(string str)
{
    float n;
    n = get_float("%s", str);
    if (n < 0)
    {
        return n = get_unsigned_float(str);
    }
    else
    {
        return n;
    }
}

int minimize(int cent)
{
    int result, quaters, dimes, nickels, pennies; 
    result = cent;

    quaters = number_of_coins(result, 25);
    result = cent % 25;
    dimes = number_of_coins(result, 10);
    result = result % 10;
    nickels = number_of_coins(result, 5);
    result = result % 5;
    pennies = number_of_coins(result, 1);
    result = result % 1;
    return pennies + nickels + dimes + quaters;
}

int number_of_coins(int cash, int coin)
{
    int r = cash % coin;
    int coins = (cash - r) / coin;
    return coins;
}

File: psets/sentimental/cash/cash.py
from cs50 import get_float


def getnumber():
    num = get_float("Change owed: ")
    if num <= 0:
        return getnumber()
    elif num is None:
        return getnumber()
    else:
        return float(num)


cash = getnumber()
cash *= 100
res = 0

quaters, dimes, nickels, pennies = 0, 0, 0, 0

while cash != 0:
    if cash / 25 >= 1:
        quaters = (cash - (cash % 25)) / 25
        cash = cash - quaters * 25
    elif cash / 10 >= 1:
        dimes = (cash - (cash % 10)) / 10
        cash = cash - dimes * 10
    elif cash / 5 >= 1:
        nickels = (cash - (cash % 5)) / 5
        cash = cash - nickels * 5
    elif cash / 1 >= 1:
        pennies = (cash - (cash % 1)) / 1
        cash = cash - pennies * 1
    else:
        break

res = quaters + dimes + nickels + pennies
print(f"{int(res)}")